Parsely/wp-parsely
# Add conditional for CPT archives and CPT term archives

## Description
When you register a custom post type with a taxonomy, the archive pages don't have the right metadata, they pick up the metadata from the first item in the loop instead of the main archive page.

What I'm trying here is adding the `is_post_type_archive()` conditional and switching to getting the queried object instead of just calling `get_the_category()` as that doesn't work in all cases.

Edit:  I found that there's another case that I wanted to target: custom post type taxonomy archives.  I've addressed that and added unit tests for that as well. 

Fixes #327 

## Motivation and Context
Fixing this for custom post type archives

## How Has This Been Tested?
1. On your test site, make sure you have a CPT with a category
2. Also make sure you have a regular post type with a category
3. Check this PR out to your test site
4. Go to the Archive pages for both the CPT and the regular post category 
5. Check to see if the right metadata is present for them both.


## Types of changes


Question:  Should we also check if the post type is currently added to the Parse.ly options?
